frontends/web/src/routes/device/bitbox02/wizard.tsx

```diff
@@ -14,211 +14,154 @@
 
-import { Component } from 'react';
-import { getStatus, getVersion, VersionInfo, verifyAttestation, TStatus } from '../../../api/bitbox02';
+import { useEffect, useState } from 'react';
+import { useLoad, useSync } from '../../../hooks/api';
+import { getStatus, getVersion, verifyAttestation } from '../../../api/bitbox02';
 import { attestationCheckDone, statusChanged } from '../../../api/devicessync';
-import { UnsubscribeList, unsubscribe } from '../../../utils/subscriptions';
 import { route } from '../../../utils/route';
 import { AppUpgradeRequired } from '../../../components/appupgraderequired';
 import { FirmwareUpgradeRequired } from '../upgrade-firmware-required';
 import { Main } from '../../../components/layout';
 import { Unlock } from './unlock';
 import { Pairing } from './setup/pairing';
-import { Wait } from './setup/wait';
 import { SetupOptions, TWalletCreateOptions, TWalletSetupChoices } from './setup/choose';
 import { CreateWallet } from './setup/wallet-create';
 import { RestoreFromSDCard, RestoreFromMnemonic } from './setup/wallet-restore';
 import { CreateWalletSuccess, RestoreFromMnemonicSuccess, RestoreFromSDCardSuccess } from './setup/success';
 
-type Props = {
+type TProps = {
   deviceID: string;
 }
 
-type State = {
-    versionInfo?: VersionInfo;
-    attestation: boolean | null;
-    status: '' | TStatus;
-    appStatus: '' | TWalletSetupChoices;
-    createOptions?: TWalletCreateOptions;
-    // if true, we just pair and unlock, so we can hide some steps.
-    unlockOnly: boolean;
-    showWizard: boolean;
-    waitDialog?: {
-        title: string;
-        text?: string;
-    };
-}
-
-export class Wizard extends Component<Props, State> {
-  constructor(props: Props) {
-    super(props);
-    this.state = {
-      attestation: null,
-      status: '',
-      appStatus: '',
-      unlockOnly: true,
-      showWizard: false,
-      waitDialog: undefined,
-    };
-  }
-
-  private unsubscribeList: UnsubscribeList = [];
-
-  public componentDidMount() {
-    const { deviceID } = this.props;
-    getVersion(deviceID).then(versionInfo => {
-      this.setState({ versionInfo });
-    });
-    this.updateAttestationCheck();
-    this.onStatusChanged();
-    this.unsubscribeList = [
-      statusChanged(deviceID, this.onStatusChanged),
-      attestationCheckDone(deviceID, this.updateAttestationCheck),
-    ];
-  }
-
-  private updateAttestationCheck = () => {
-    verifyAttestation(this.props.deviceID).then(attestation => {
-      this.setState({ attestation });
-    });
-  };
-
-  private handleGetStarted = () => {
-    this.setState({ status: '' });
+export const Wizard = ({ deviceID }: TProps) => {
+  const versionInfo = useLoad(() => getVersion(deviceID));
+  const attestation = useSync(
+    () => verifyAttestation(deviceID),
+    cb => attestationCheckDone(deviceID, () => {
+      verifyAttestation(deviceID).then(cb);
+    })
+  );
+  const [appStatus, setAppStatus] = useState<'' | TWalletSetupChoices>('');
```

Comment:
Since you are here, wdyt about renaming appStatus to something else, i.e. setupChoice or similar?

Reply:
`appStatus` is named on contrast to `status` which is the device status.

no strong opinion, but i don't feel like spending time on it :see_no_evil: